packages/components/src/internal/components/domainproperties/dataclasses/models.ts

```diff
@@ -88,8 +99,42 @@ export class DataClassModel implements DataClassModelConfig {
         );
     }
 
+    parentAliasInvalid(alias: Partial<IParentAlias>): boolean {
```

Comment:
looks like `parentAliasInvalid` and `getDuplicateAlias` are duplicates from the SampleTypeModel. Do you think there is a way this code can be shared?